File: cardioCare-API/src/main/java/cl/ufro/dci/cardiocare/message/service/MessageService.java
package cl.ufro.dci.cardiocare.message.service;

import cl.ufro.dci.cardiocare.message.dto.*;

import java.util.List;

public interface MessageService {
    MessageResponse send(MessageRequest request);
    List<MessageResponse> getByConsultation(Long consultationId);
    List<MessageResponse> getInbox(Long receiverId);
    List<MessageResponse> getSent(Long senderId);
}

File: cardioCare-UI/cardioCare/src/app/messages/services/message.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { MessageRequest, MessageResponse } from '../models/message.model';

@Injectable({
  providedIn: 'root'
})
export class MessageService {

  private apiUrl = 'http://localhost:8080/api/v1/messages';

  constructor(private http: HttpClient) { }

  send(request: MessageRequest): Observable<MessageResponse> {
    return this.http.post<MessageResponse>(this.apiUrl, request);
  }

  getByConsultation(id: number): Observable<MessageResponse[]> {
    return this.http.get<MessageResponse[]>(`${this.apiUrl}/consultation/${id}`);
  }

  getInbox(userId: number): Observable<MessageResponse[]> {
    return this.http.get<MessageResponse[]>(`${this.apiUrl}/inbox/${userId}`);
  }

  getSent(userId: number): Observable<MessageResponse[]> {
    return this.http.get<MessageResponse[]>(`${this.apiUrl}/sent/${userId}`);
  }
}
